test/test_grid_search.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import fairlearn.grid_search as gs
+
+import numpy as np
+import pandas as pd
+
+import pytest
+
+class LeastSquaresLearner:
+    def __init__(self):
+        self.weights = None
+
+    def fit(self, X, Y, sample_weight):
+        sqrtW = np.sqrt(sample_weight)
+        matX = np.array(X) * sqrtW[:, np.newaxis]
+        vecY = Y * sqrtW
+        self.lsqinfo = np.linalg.lstsq(matX, vecY, rcond=-1)
+        self.weights = pd.Series(self.lsqinfo[0], index=list(X))
+
+    def predict(self, X):
+        pred = X.dot(self.weights)
+        return 1*(pred > 0.5)
+
+class TestGridSearch:
+    def test_generate_p0_p1_smoke(self):
+        Y = [int(x) for x in '0110101' '0111101' '001011']
```

Comment:
What is this meant to do? Is the formatting as 0/1 sequences helpful in understanding it? If so, why are we converting it to int. I think it would be very helpful to have a comment explaining it since it's all over this file.

Reply:
Have improved